src/test/kotlin/no/skatteetaten/aurora/boober/feature/S3StorageGridFeatureTest.kt

```diff
@@ -0,0 +1,282 @@
+                }
+           }"""
+            )
+        }.singleApplicationError("objectArea name=default used 2 times for same application")
+    }
+
+    @Test
+    fun `verify setting tenant in AuroraConfig is overridden with default`() {
+        val areaName = "default"
+        val spec = createAuroraDeploymentContext(
+            """{ 
+                "s3": {
+                    "$areaName" : {
+                        "bucketName" : "anotherId",
+                        "tenant": "demo-utv"
+                    }
+                }
+           }"""
+        ).spec
+        assertThat(spec.s3ObjectAreas.find { it.area == areaName }?.tenant).isEqualTo("paas-utv")
+    }
+
+    @Test
+    fun `verify is able to disable s3 when simple config`() {
+
+        verify(exactly = 0) { provisioner.getOrProvisionCredentials(any(), any()) }
+
+        generateResources(
+            """{ 
+                "s3": false
+           }""",
+            createdResources = 0,
+            resources = mutableSetOf(createEmptyApplicationDeployment(), createEmptyDeploymentConfig())
+        )
+    }
+
+    @Test
+    fun `verify is able to disable s3 when expanded config`() {
+
+        every {
+            provisioner.getOrProvisionCredentials(any(), match {
+                it.size == 1 && it.find { it.bucketPostfix == bucket1Name && it.objectAreaName == area1Name } != null
+            })
+        } returns listOf(sgRequestsWithCredentials(area1Name, bucket1Name))
+
+        val resources = generateResources(
+            """{ 
+                "s3": {
+                    "$area1Name" : {
+                        "bucketName": "$bucket1Name"
+                    },
+                    "$area2Name" : {
+                        "enabled": false,
+                        "bucketName" : "anotherId"
+                    }
+                }
+           }""",
+            createdResources = 1,
+            resources = mutableSetOf(createEmptyApplicationDeployment(), createEmptyDeploymentConfig())
+        )
+
+        resources.verifyS3SecretsAndEnvs(listOf(area1Name))
+    }
+
+    @Test
+    fun `verify two objectareas with different bucketnames`() {
+        val bucket2Name = "anotherBucket"
+
+        every {
+            provisioner.getOrProvisionCredentials(any(), match { r ->
+                r.run {
+                    find { it.bucketPostfix == bucket1Name && it.objectAreaName == area1Name } != null
+                            && find { it.bucketPostfix == bucket2Name && it.objectAreaName == area2Name } != null
+                }
+            })
+        } returns listOf(
+            sgRequestsWithCredentials(area1Name, bucket1Name),
+            sgRequestsWithCredentials(area2Name, bucket2Name)
+        )
+
+        val resources = generateResources(
+            """{ 
+                "s3Defaults": {
+                    "bucketName": "$bucket1Name"
+                },
+                "s3": {
+                    "$area1Name" : { },
```

Comment:
Funker det med tom {}? Hvordan ser det ut i yaml?
Burde vi heller ha noe mer eksplisitt, slik som ```"$area1Name": true```

Reply:
Litt usikker på hvordan dette vil se ut i yaml, men enabled: true er default, så det er derfor det funker med tom blokk. Hvis vi vil ha støtte for "simple config" her, så tenker jeg en egen oppgave på det, så vi får avsluttet.

Reply:
enig i det. Var egentlig bare en forundring og tanke på en liten utvidelse